instrumentation/jaxrs/jaxrs-common/testing/src/main/java/io/opentelemetry/instrumentation/jaxrs/AbstractJaxRsFilterTest.java

```diff
@@ -0,0 +1,191 @@
+        Arguments.of(
+            "/test3/hi/bob", false, false, "/test3/hi/{name}", "Test3.hello", "Test3 bob!"),
+        // Resteasy and Jersey give different resource class names for just the below case
+        // Resteasy returns "SubResource.class"
+        // Jersey returns "Test1.class
+        // Arguments.of("/test/hello/bob", true, false, "/test/hello/{name}", "Test1.hello",
+        // "Aborted"),
+        Arguments.of(
+            "/test2/hello/bob", true, false, "/test2/hello/{name}", "Test2.hello", "Aborted"),
+        Arguments.of("/test3/hi/bob", true, false, "/test3/hi/{name}", "Test3.hello", "Aborted"),
+        Arguments.of(
+            "/test/hello/bob",
+            false,
+            true,
+            null,
+            "PrematchRequestFilter.filter",
+            "Aborted Prematch"),
+        Arguments.of(
+            "/test2/hello/bob",
+            false,
+            true,
+            null,
+            "PrematchRequestFilter.filter",
+            "Aborted Prematch"),
+        Arguments.of(
+            "/test3/hi/bob",
+            false,
+            true,
+            null,
+            "PrematchRequestFilter.filter",
+            "Aborted Prematch"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("requests")
+  void request(
+      String resource,
+      boolean abortNormal,
+      boolean abortPrematch,
+      String route,
+      String controllerName,
+      String expectedResponse)
+      throws Exception {
+    Assumptions.assumeTrue(!abortPrematch || testAbortPrematch());
+
+    setAbortStatus(abortNormal, abortPrematch);
+    boolean abort = abortNormal || abortPrematch;
+
+    TestResponse response = runRequest(resource);
+    assertThat(response.text).isEqualTo(expectedResponse);
+
+    assertThat(response.status)
+        .isEqualTo(
+            abort
+                ? 401 // Response.Status.UNAUTHORIZED.statusCode
+                : 200); // Response.Status.OK.statusCode
+
+    String serverRoute = route != null ? route : defaultServerRoute();
+    String method = runsOnServer() ? "POST" : "GET";
+    String expectedServerSpanName = serverRoute == null ? method : method + " " + serverRoute;
+    testing()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> {
+                      span.hasName(expectedServerSpanName).hasKind(SpanKind.SERVER);
+                      if (runsOnServer() && abortNormal) {
+                        span.hasStatus(StatusData.unset());
+                      }
+                    },
+                    span -> {
+                      span.hasName(controllerName).hasParent(trace.getSpan(0));
+                      if (abortPrematch) {
+                        span.hasAttributesSatisfyingExactly(
+                            satisfies(
+                                CodeIncubatingAttributes.CODE_NAMESPACE,
```

Comment:
consider static `CODE_FUNCTION` and `CODE_NAMESPACE`

```suggestion
                               CODE_NAMESPACE,
```